data/lua/on_event.lua

```diff
@@ -2,54 +2,33 @@
 -- so you have to call this function from a toplevel lua tag or from a preload event.
 -- It is also not possible to use this for first_time_only=yes events.
 
-if rawget(_G, "core_on_event") then
-	return rawget(_G, "core_on_event")  -- prevent double execution
-end
-
-local event_handlers = {}
-
-local old_on_event = wesnoth.game_events.on_event or function(eventname) end
-wesnoth.game_events.on_event = function(eventname)
-	old_on_event(eventname)
-	local context = nil
-	for _, entry in pairs(event_handlers[eventname] or {}) do
-		if context == nil then
-			context = wesnoth.current.event_context
-		end
-		entry.h(context)
-	end
-end
+-- This api used to be the default way to add events from lua (with its own implementation in lua
+-- based on game_events.on_event). Now its just calls game_evets.add, in particular because otherwise the
+-- priority parammter wouldn't work accord the differnt implmentaions..
```

Comment:
This line is full of typos and also doesn't quite make sense.

Reply:
Using "across" made the intended meaning clear, but please also correct the other spellings:

parameter
different
implementations